products/template/build_scripts/export.py

```diff
@@ -28,20 +28,15 @@
 def generate_data_dictionaries():
     org_metadata = OrgMetadata.from_path(Path(PRODUCT_METADATA_REPO_PATH))
 
-    dataset_metadata_path = PRODUCT_PATH / "data_dictionary.yml"
-    metadata = org_metadata.product("template_db").dataset("template_db")
-    metadata.write_to_yaml(dataset_metadata_path)
-
-    html_path = PRODUCT_PATH / "data_dictionary.html"
-    pdf_writer.generate_html_from_yaml(
-        dataset_metadata_path,
-        html_path,
-        pdf_writer.DEFAULT_DATA_DICTIONARY_TEMPLATE_PATH,
-        pdf_writer.DEFAULT_DATA_DICTIONARY_STYLESHEET_PATH,
+    yaml_writer.write_yaml(
```

Comment:
Nice, this is much cleaner. It would be nice to get this sort of logic into `lifecycle/package` (as in, outputs are configured within metadata, so we just need something like `metadata.generate_attachments`) but that certainly seems like a separate issue

Reply:
definitely wanna move this stuff into `lifecycle` next

Reply:
yeah, def plus +1 to moving this to [packaging artifacts](https://github.com/NYCPlanning/product-metadata/blob/main/packaging/artifacts.yml). I'm foreseeing dynamic dispatch based on the `artifact.type`, which would make it extensible for the outside world.

Reply:
Was initially hesitant about a separate module for just writing the dataset yml, but I think it really fits with that approach 👆. Though I do think it's probably less of a general `yaml_writer` and more of a `dataset_yaml_writer` or similar. We should probably align on a standard interface for these methods that are writing artifacts... maybe 
`.generate(dataset_md, out_path, **kwargs)`